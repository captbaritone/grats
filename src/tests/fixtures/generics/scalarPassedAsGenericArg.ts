/** @gqlType */
type Wrapper<T> = {
  /** @gqlField */
  value: T;
};

/** @gqlType */
type OtherType = {
  /** @gqlField */
  wrapper: Wrapper<string>;
};
