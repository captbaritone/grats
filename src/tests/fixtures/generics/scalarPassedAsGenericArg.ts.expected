-----------------
INPUT
----------------- 
/** @gqlType */
type Wrapper<T> = {
  /** @gqlField */
  value: T;
};

/** @gqlType */
type OtherType = {
  /** @gqlField */
  wrapper: Wrapper<string>;
};

-----------------
OUTPUT
-----------------
src/tests/fixtures/generics/scalarPassedAsGenericArg.ts:10:20 - error: Expected `Wrapper` to be passed a GraphQL type argument for type parameter `T`.

10   wrapper: Wrapper<string>;
                      ~~~~~~

  src/tests/fixtures/generics/scalarPassedAsGenericArg.ts:2:14
    2 type Wrapper<T> = {
                   ~
    Type parameter `T` is defined here
  src/tests/fixtures/generics/scalarPassedAsGenericArg.ts:4:10
    4   value: T;
               ~
    and expects a GraphQL type because it was used in a GraphQL position here.
